src/server/lib/__tests__/headers.test.ts

```diff
@@ -0,0 +1,74 @@
+import { ValidRoutePathsArray } from '@/shared/model/Routes';
+import Redis from 'ioredis-mock';
+import request from 'supertest';
+import { getServerInstance } from './sharedConfig';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+
+// Override the default 5s max timeout for these tests because Supertest takes some time to run.
+jest.setTimeout(10000);
+
+describe('Content Security Policy headers', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetModules();
+    jest.mock('aws-sdk');
+    jest.mock('@/server/lib/getAssets', () => ({
+      getAssets: () => ({
+        main: { js: 'mocked' },
+        vendors: { js: 'mocked' },
+        runtime: { js: 'mocked' },
+      }),
+    }));
+
+    jest.mock('@/server/lib/serverSideLogger', () => ({
+      logger: {
+        info: jest.fn(),
+        error: jest.fn(),
+      },
+    }));
+
+    jest.mock('@/server/lib/redis/redisClient', () => new Redis());
+
+    jest.mock('@/server/lib/middleware/login', () => ({
+      loginMiddleware: jest.fn((req, res, next) => next()),
+    }));
+    jest.mock('@/server/lib/IDAPIFetch');
+  });
+
+  it('should return CSP headers for routes', async () => {
+    const { server, teardownServer } = await getServerInstance();
+    const { baseUri, apiDomain } = getConfiguration();
+    for (const route of ValidRoutePathsArray) {
+      const response = await request(server).get(route);
+      // Is the CSP header set?
+      expect(response.header).toHaveProperty('content-security-policy');
+      const splitCSPHeader =
+        response.header['content-security-policy'].split(';');
+      // Does the CSP header match what we expect it to be? If the CSP settings
+      // in Helmet are updated, we expect to need to update these tests.
+      expect(splitCSPHeader).toContain("base-uri 'none'");
+      expect(splitCSPHeader).toContain("default-src 'none'");
+      expect(splitCSPHeader).toContain(
+        `script-src ${baseUri} 'sha256-411aj9j2RJj78RLGlCL/KDMK0fe6OEh8Vp6NzyYIkP4=' www.google-analytics.com sourcepoint.theguardian.com gdpr-tcfv2.sp-prod.net ccpa.sp-prod.net ccpa-service.sp-prod.net ccpa-notice.sp-prod.net cdn.privacy-mgmt.com www.google.com www.gstatic.com assets.guim.co.uk 'unsafe-eval'`,
+      );
+      expect(splitCSPHeader).toContain(
+        `img-src ${baseUri} static.guim.co.uk ophan.theguardian.com www.google-analytics.com www.google.com`,
+      );
+      expect(splitCSPHeader).toContain('font-src assets.guim.co.uk');
+      expect(splitCSPHeader).toContain(
+        `connect-src www.google-analytics.com vendorlist.consensu.org consent-logs.${apiDomain} sourcepoint.theguardian.com gdpr-tcfv2.sp-prod.net ccpa.sp-prod.net ccpa-service.sp-prod.net ccpa-notice.sp-prod.net cdn.privacy-mgmt.com api.nextgen.guardianapps.co.uk https://api.pwnedpasswords.com localhost:1234 www.google.com o14302.ingest.sentry.io`,
```

Comment:
i wonder if the two longer strings (this one and the one on line 52) could be constructed from arrays that are then joined into the final string?

suggesting because some of them are sourced from our mocked env object and would be good to assert against the values stored there so when the env object is changed this changes too.. also it might make it a more pleasant experience to read/modify them in the future :)